Sources/Common/OperaDecodable.swift

```diff
@@ -54,3 +55,18 @@ import Foundation
 public protocol OperaDecodable {
     static func decode(_ json: Any) throws -> Self
 }
+
+public extension OperaDecodable where Self: Decodable {
+    
+    static func decode(_ json: Any) throws -> Self {
+        
+        let decoder = JSONDecoder();
+        let dateFormatter = DateFormatter()
+        dateFormatter.calendar = Calendar.current
+        dateFormatter.timeZone = TimeZone.current
+        dateFormatter.dateFormat = DateFormats.iso8601
```

Comment:
The date format should be customizable because it is different in every app. I am not even sure if this would support different modes of ISO 8601. What if we add an optional `DateFormatter` variable to the protocol?

Reply:
agree! it should be customizable